src/sentry/integrations/opsgenie/integration.py

```diff
@@ -120,7 +120,46 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:  # type:ignore
 
 
 class OpsgenieIntegration(IntegrationInstallation):
-    pass
+    def get_client(self) -> Any:
+        return OpsgenieClient(integration=self.model)
+
+    def get_organization_config(self) -> Sequence[Any]:
+        fields = [
+            {
+                "name": "team_table",
+                "type": "table",
+                "label": "Opsgenie teams with the Sentry integration enabled",
+                "help": "If teams need to be updated, deleted, or added manually please do so here. Alert rules will need to be individually updated for any additions or deletions of teams.",
+                "addButtonText": "",
+                "columnLabels": {"team": "Team", "integration_key": "Integration Key"},
+                "columnKeys": ["team", "integration_key"],
+                "confirmDeleteMessage": "Any alert rules associated with this team will stop working. The rules will still exist but will show a `removed` team.",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        client = self.get_client()
+        # get the team ID/test the API key for a newly added row
+        teams = data["team_table"]
+        for team in teams:
+            if team["id"] != "":
+                continue
```

Comment:
I think it might read a little easier if instead of iterating through all teams, you sift through only the `id-less` ones as a variable first:

```python
unsaved_teams = [team for team in teams if team["id"] != ""]
for team in unsaved_teams:
  # make api req
```

It's no different logically, but makes explicit what we're checking for with the condition.